dhis2/dhis2-core
# feat: api endpoint for user impersonation feature

## Summary
Converts the impersonate user feature implemented in the `SwitchUserFilter` into a dedicated API endpoint, defined in the new `ImpersonateUserController`.

### Automated tests
`ImpersonateUserControllerTest`

### Manual test

1.  Capture the cookie in the response, from the `Set-Cookie` response header.
```
curl -X GET "http://localhost:9090/api/me?fields=username%2Cid" \
    --basic --user admin:district
```
2. Use the cookie in next requests.
```
curl -X GET "http://localhost:9090/api/me?fields=username%2Cid" \
    -H "Cookie: JSESSIONID=SAME_COOKIE.node0;"
```
3. Do the impersonate request.
```
curl -X POST "http://localhost:9090/api/auth/impersonate?username=spanishB" \
    -H "Cookie: JSESSIONID=SAME_COOKIE.node0;"
```
4. Observe, you have become the impersonated user.
```
curl -X GET "http://localhost:9090/api/me?fields=username%2Cid" \
    -H "Cookie: JSESSIONID=SAME_COOKIE.node0;"
```
